ui/pages/login.tsx

```diff
@@ -299,13 +308,77 @@ const Login: NextPage = () => {
         text: "Invalid login method",
       });
     }
+    if (isSignInWithHardwareKey(node) && isSequencedLogin) {
+      node.meta.label.text = "Sign in using Passkey";
+    }
     return node;
   });
 
+  // automatically forward to single oidc provider if it is the only option
+  const isSingleOidcOption =
+    isSequencedLogin &&
+    renderFlow?.ui.nodes.length === 2 &&
+    renderFlow?.ui.nodes[1].group === "oidc" &&
+    (renderFlow?.ui.nodes[0].attributes as UiNodeInputAttributes).name ===
+      "csrf_token";
+  if (isSingleOidcOption) {
```

Comment:
How about a function with this logic so the if can become:

```typescript
  // outside of the component
  function isSingleOidcOption(isSequencedLogin boolean, renderFlow renderFlowType) {
      return isSequencedLogin &&
          renderFlow?.ui.nodes.length === 2 &&
          renderFlow?.ui.nodes[1].group === "oidc" &&
          (renderFlow?.ui.nodes[0].attributes as UiNodeInputAttributes).name === "csrf_token";
  }
  
  ...
  
  if (isSingleOidcOption(isSequencedLogin, renderFlow)) {
      ...
   }
```

Reply:
I usually try to avoid callbacks when the logic can be directly evaluated. It seems simpler and a bit cheaper to me.